### ISSUE TITLE
[Train] Ray Train not respecting fractional resource requests and label selectors

### ISSUE THREAD

**Issue by jleben** [ray-project/ray#63241]
### What happened + What you expected to happen

In addition to creating placement groups (with correct resource requests and label selectors) Ray Train in 2.55.1 also uses `request_resources()`. However, the latter has 2 issues:

- It rounds up fractional resources to a whole number.
- It does not respect the user provided label selectors

Here is a screenshot of from Ray Dashboard right after launching a job as described in the reproduction section below:

<img width="1165" height="243" alt="Image" src="https://github.com/user-attachments/assets/e5d2e7ea-a710-4018-a06e-8296eef7e724" />

I would expect Ray to only allocate 0.1 CPU from a worker group providing m6i.xlarge nodes, but you can see that `request_resources()` allocates another 1 CPU which triggers the scaling up of a a different worker group.

This can potentially cause huge resource overallocation when launching training jobs with small fractional resources. For example, it prevents me from efficiently running many light training jobs (evaluating a model on many benchmarks or hyperparameter tuning) where each job uses only a fraction of a GPU.

### Versions / Dependencies

Ray 2.55.1

### Reproduction script

Create a Ray cluster with 2 worker groups:

- One providing m6i.xlarge EC2 instances, with a label "instance-type: m6i.xlarge"
- One providing m6i.large EC2 instances

Submit a job with the following entrypoint:

```
import ray
import ray.train
import ray.train.torch
import time

ray.init()

def train_func():
    time.sleep(60 * 10)

ray.train.torch.TorchTrainer(
    train_func,
    scaling_config=ray.train.ScalingConfig(
        num_workers=1,
        resources_per_worker={"CPU": 0.1},
        label_selector={"instance-type": "m6i.xlarge"},
    ),
).fit()
```

### Issue Severity

High: It blocks me from completing my task.

**Comment by liulehui** [comment]
Thanks for reporting! should be fixed by the these 2 PRs:
https://github.com/ray-project/ray/pull/63287
https://github.com/ray-project/ray/pull/63306

feel free to try out in HEAD.
